### PR TITLE
remove block_until_ready from transfer_and_register_cpu_chunk

### PR DESCRIPTION
**Pull request by saikat-royc**
# Description

Start with a short description of what the PR does and how this is a change from
the past.

   Eliminates the second synchronous `block_until_ready()` call in the KV
    offload pipeline, enabling fully non-blocking Device-to-Host (D2H)
    transfers.

The rest of the description includes relevant details and context, examples:
- why is this change being made,
- the problem being solved and any relevant context,
- why this is a good solution,
- some information about the specific implementation,
- shortcomings of the solution and possible future improvements.

     
    Key changes:
    - Asynchronous Dispatch: Modified `_transfer_and_register_cpu_chunks` to
      return a list of JAX futures (unready arrays) and removed the
      `jax.block_until_ready()` stall.
    - Soft Budgeting Polling: Updated `_process_completed_saves` in the main
      feedback loop to implement a completion check:
      - Jax Array Readiness: Performs a non-blocking hardware check
         (`array.is_ready()`) to confirm the TPU finished moving the bytes.
    -  The signal to the Scheduler to "Free" staging buffer slots is now strictly gatekept by cpu chunks readiness.
    - Unit test for async save + is_ready() check for the jax array stored
      in cpu backend

If the change fixes a Github issue, please include a link, e.g.,:
FIXES: #123456

# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

Manual runs of the kv host offload unit tests in v7 machine

# Checklist

Before submitting this PR, please make sure:
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

**Review by juncgu-google — APPROVED** [review]
LGTM

**Comment by saikat-royc** [comment]
b/499150571 for detailed traces
